app/models/notification.rb

```diff
@@ -35,6 +33,11 @@ def self.logger(extra_fields = {})
 
   scope :due_today, -> { where(remind_on: Date.current, status: [:pending]) }
 
+  def record_communication(communication)
```

Comment:
It seems a bit strange that this responsibility lives on the notification model. The channel creates the detail and communication, right? Can the dispatch service orchestrate this "linkage"?